### PR TITLE
ci: support releasing proto definitions to BSR

### PR DESCRIPTION
**Pull request by rach-id**
Adds support for pushing the proto definitions to BSR

## Summary by CodeRabbit

- **New Features**
	- Introduced a continuous integration (CI) workflow for automated checks on Protocol Buffers.
	- Added a new module configuration for managing protocol buffers.

- **Documentation**
	- New workflow file created to outline CI processes.
	- Configuration file established for module versioning and structure.

### CODE REVIEW

File: .github/workflows/buf-ci.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Buf CI
```
Comment (rootulp) [inline .github/workflows/buf-ci.yaml:1]: [nit] 

So the workflow name matches the file name
```suggestion
name: buf-ci
```

File: .github/workflows/buf-ci.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: bufbuild/buf-action@v1
+        with:
+          token: ${{ secrets.BUF_TOKEN }}
+          input: "pb"
+          breaking-against: 'https://github.com/celestiaorg/nmt.git#branch=main,ref=HEAD~1,subdir=pb'
+          # Run breaking change, lint, and format checks for Protobuf sources against all branches,
+          # 'pb' # subdirectory, then push to the BSR once validated
```
Comment (rootulp) [inline .github/workflows/buf-ci.yaml:26]: This line doesn't make sense to me. 

1. Why two `#`?
2. What is 'pb'?
3. Can this use the longer format Buf Schema Registry for the first usage of BSR. So something like "then push to the Buf Schema Registry (BSR) once validated

**Comment by rach-id** [comment]
CI will be fixed by:
https://github.com/celestiaorg/nmt/pull/276 
and 
https://github.com/celestiaorg/nmt/issues/275

**Review by rootulp — COMMENTED** [review]
How was this PR tested? Did your fork push proto definitions to Buf Schema Registry successfully? Or will it be tested after this PR merges?

File: .github/workflows/buf-ci.yaml (hunk 2 above)
Comment (rach-id) [reply .github/workflows/buf-ci.yaml]: 1. [762e67b](https://github.com/celestiaorg/nmt/pull/274/commits/762e67bacd69a96fbaf757bf1034014109ff476b)
2. folder containing the protos
3. [3fb737f](https://github.com/celestiaorg/nmt/pull/274/commits/3fb737f7cb7c4c087743684955a2c48e0b767734)

**Comment by rach-id** [comment]
> How was this PR tested? Did your fork push proto definitions to Buf Schema Registry successfully? Or will it be tested after this PR merges?

it should be tested after the PR is merged

**Review by rootulp — APPROVED** [review]

File: .github/workflows/buf-ci.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: buf-ci
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
```
Comment (MSevey) [inline .github/workflows/buf-ci.yaml:7]: unless you are creating a bunch of tags that start with v that aren't version tags, I think you can simplify this to just `v*`.

**Comment by MSevey** [comment]
> > How was this PR tested? Did your fork push proto definitions to Buf Schema Registry successfully? Or will it be tested after this PR merges?
> 
> it should be tested after the PR is merged

For GH actions that can't be tested on the PR, because they require being on main to be tested, I recommend either using a fork or having a personal repo like `workflows` where you can do all testing on the action to make sure it works. 

I have a `MSevey/workflows` repo or example that I do a lot of testing on for new actions to better understand them prior to merging into a project.

**Review by MSevey — APPROVED** [review]

**Comment by rach-id** [comment]
I tried running this in a personal repo but I keep hitting certain upload permission errors. I will merge this one, test it. if it works, there is no need to waste more time on it. If not, I will fix in a follow up PR

**Pull request merged**